packages/foundation-models/package.json

```diff
@@ -22,13 +22,14 @@
   "scripts": {
     "compile": "tsc",
     "compile:cjs": "tsc -p tsconfig.cjs.json",
-    "test": "NODE_OPTIONS=--experimental-vm-modules jest",
+    "test": "pnpm generate-zod && NODE_OPTIONS=--experimental-vm-modules jest",
     "lint": "eslint \"**/*.ts\" && prettier . --config ../../.prettierrc --ignore-path ../../.prettierignore -c",
     "lint:fix": "eslint \"**/*.ts\" --fix && prettier . --config ../../.prettierrc --ignore-path ../../.prettierignore -w --log-level error",
     "check:public-api": "node --loader ts-node/esm ../../scripts/check-public-api-cli.ts",
     "generate:azure-openai": "openapi-generator --generateESM --clearOutputDir -i ./src/azure-openai/spec/inference.yaml -o ./src/azure-openai/client --schemaPrefix AzureOpenAi",
     "postgenerate:azure-openai": "rm ./src/azure-openai/client/inference/*.ts && pnpm lint:fix",
-    "generate-zod": "ts-to-zod src/azure-openai/azure-openai-embedding-types.ts src/azure-openai/ts-to-zod/azure-openai-embedding-types.zod.ts"
+    "generate-zod-config": "node --loader ts-node/esm ../../scripts/generate-zod-config.ts './src/azure-openai/client/inference/schema/*.ts' ./src/azure-openai/ts-to-zod",
```

Comment:
[q] why is the input in quotes but the output not?

Reply:
If I remove the quote, I will get all satisfied files in the argument list according to the glob pattern. Instead, I am using quote to make it a pure string and then parse the file paths in code.

Reply:
The string version doesn't work for me.

Reply:
Also, the generated ts-to-zod is not deleted post tests

Reply:
> Also, the generated ts-to-zod is not deleted post tests

I am not sure if it is necessary also to delete the generated zod.ts? That is also why I git ignored these files. Could be that someone want to have a look into them?

Reply:
> The string version doesn't work for me.

I feel this is a problem for windows due to slash and back slash.. Anyways, I now parse the list directly in the arguments and removed the need for single quote.

Reply:
This still doesn't work unfortunately. The glob pattern is not evaluated when you run the command, so in the code, I still just see the string and not the list of files. Can we just pass the directory and get all files ?

Reply:
> Could be that someone want to have a look into them?

We are regenerating it everytime anyway, so why keep a version lying around (to maintain this, we then have to add a .gitignore, eslint config). Having a cleanup step would be better IMO.
The chance that someone would actually go through this code is really low. In case they want to, they can comment out this cleanup step and debug.

Reply:
I pushed the solution, which is using `\"` instead of `'` or ``. And I have to revert to the previous variant to let glob lib parse the input within the script. On Windows, node will not parse glob pattern automatically.

Reply:
> > Could be that someone want to have a look into them?
> 
> We are regenerating it everytime anyway, so why keep a version lying around (to maintain this, we then have to add a .gitignore, eslint config). Having a cleanup step would be better IMO. The chance that someone would actually go through this code is really low. In case they want to, they can comment out this cleanup step and debug.

Removed.